src/storage/remote/storage.c

```diff
@@ -141,30 +194,37 @@ storageRemoteInfoList(
         // Send command
         protocolClientWriteCommand(this->client, command);
 
-        // Read list.  The list ends when there is a blank line -- this is safe even for file systems that allow blank filenames
-        // since the filename is json-encoded so will always include quotes.
+        // Read list
+        PackRead *read = pckReadNew(protocolClientIoRead(this->client));
+        StorageRemoteInfoParseData parseData = {.read = read};
+
         MEM_CONTEXT_TEMP_RESET_BEGIN()
         {
-            const String *name = protocolClientReadLine(this->client);
+            pckReadArrayBeginP(read);
 
-            while (strSize(name) != 0)
+            while (pckReadNext(read))
             {
-                StorageInfo info = {.exists = true, .level = level, .name = jsonToStr(name)};
+                pckReadObjBeginP(read);
 
-                storageRemoteInfoParse(this->client, &info);
+                StorageInfo info = {.exists = true, .level = level, .name = pckReadStrP(read)};
+
+                storageRemoteInfoParse(&parseData, &info);
                 callback(callbackData, &info);
 
                 // Reset the memory context occasionally so we don't use too much memory or slow down processing
                 MEM_CONTEXT_TEMP_RESET(1000);
 
-                // Read the next item
-                name = protocolClientReadLine(this->client);
+                pckReadObjEndP(read);
```

Comment:
Why is pckReadObjEndP(read);  after  MEM_CONTEXT_TEMP_RESET(1000); instead of before?

Reply:
Because if it was before the value just read would be freed. We need it for the next iteration of the loop.